snippets/graph_neighbourhood.py

```diff
@@ -0,0 +1,68 @@
+""""""
+
+from itertools import starmap
+from operator import le, lt, and_
+from pathlib import Path
+from os import path
+from sys import argv, stdout
+from argparse import ArgumentParser
+
+from data import load_graph
+
+
+def neighbourhood(coord: tuple[int, int], borders) -> None:
+    n, s, w, e, nw, ne, sw, se = [
+        (-1, 0),
+        (1, 0),
+        (0, -1),
+        (0, 1),
+        (-1, -1),
+        (-1, 1),
+        (1, -1),
+        (1, 1),
+    ]
+    return filter(
+        lambda l: and_(*starmap(le, zip((0, 0), l)))
+        and and_(*starmap(lt, zip(l, borders))),
+        map(lambda n: tuple(map(sum, zip(coord, n))), (n, s, w, e, nw, ne, sw, se)),
```

Comment:
I would find this clearer if you wrapped this filter call in its own function whose name describes what it is doing. Then return the result of that function call.